src/jdataview.js

```diff
@@ -1,833 +1,465 @@
+	 */
+	slice(start, end, forceCopy) {
+		function normalizeOffset(offset, byteLength) {
+			return offset < 0 ? offset + byteLength : offset;
 		}
-	},
-
-	_action: function (type, isReadAction, byteOffset, littleEndian, value) {
-		return this._engineAction(
-			type,
-			isReadAction,
-			defined(byteOffset, this._offset),
-			defined(littleEndian, this._littleEndian),
-			value
-		);
-	},
 
-	_dataViewAction: function (type, isReadAction, byteOffset, littleEndian, value) {
-		// Move the internal offset forward
-		this._offset = byteOffset + dataTypes[type];
-		return isReadAction ? this._view['get' + type](byteOffset, littleEndian) : this._view['set' + type](byteOffset, value, littleEndian);
-	},
-
-	_arrayBufferAction: function (type, isReadAction, byteOffset, littleEndian, value) {
-		var size = dataTypes[type], TypedArray = global[type + 'Array'], typedArray;
-
-		littleEndian = defined(littleEndian, this._littleEndian);
+		start = normalizeOffset(start, this.byteLength);
+		end = normalizeOffset(end ?? this.byteLength, this.byteLength);
+
+		return forceCopy
+			? new jDataView(
+					this.getBytes(end - start, start, true),
+					undefined,
+					undefined,
+					this.littleEndian
+			  )
+			: new jDataView(
+					this.buffer,
+					this.byteOffset + start,
+					end - start,
+					this.littleEndian
+			  );
+	}
 
-		// ArrayBuffer: we use a typed array of size 1 from original buffer if alignment is good and from slice when it's not
-		if (size === 1 || ((this.byteOffset + byteOffset) % size === 0 && littleEndian)) {
-			typedArray = new TypedArray(this.buffer, this.byteOffset + byteOffset, 1);
-			this._offset = byteOffset + size;
-			return isReadAction ? typedArray[0] : (typedArray[0] = value);
+	/**
+	 * Aligns the pointer (clearing any bitOffset). Can also move the pointer
+	 */
+	alignBy(byteCount) {
+		this.#bitOffset = 0;
+		if ((byteCount ?? 1) !== 1) {
+			return this.skip(
+				byteCount - (this.#bytePointer % byteCount || byteCount)
+			);
 		} else {
-			var bytes = new Uint8Array(isReadAction ? this.getBytes(size, byteOffset, littleEndian, true) : size);
-			typedArray = new TypedArray(bytes.buffer, 0, 1);
-
-			if (isReadAction) {
-				return typedArray[0];
-			} else {
-				typedArray[0] = value;
-				this._setBytes(byteOffset, bytes, littleEndian);
-			}
+			return this.#bytePointer;
 		}
-	},
-
-	_arrayAction: function (type, isReadAction, byteOffset, littleEndian, value) {
-		return isReadAction ? this['_get' + type](byteOffset, littleEndian) : this['_set' + type](byteOffset, value, littleEndian);
-	},
+	}
 
-	// Helpers
+	// Setters and getters
 
-	_getBytes: function (length, byteOffset, littleEndian) {
-		littleEndian = defined(littleEndian, this._littleEndian);
-		byteOffset = defined(byteOffset, this._offset);
-		length = defined(length, this.byteLength - byteOffset);
+	#getBytes(length, byteOffset, littleEndian) {
+		littleEndian ??= this.littleEndian;
```

Comment:
Nit: all these `??=` that can be inlined as param defaults should be inlined. Let's keep the new operator only for cases where it's really necessary.